### DISCUSSION TITLE
Dssp ValueError: could not broadcast input array from shape (71,) into shape (70,)

### DISCUSSION THREAD

**Discussion by intomybioverse** [biotite-dev/biotite#621] [Q&A] [answered]
I am trying to run the biotite secondary structure analysis program in wsl to study the secondary structure of a protein after md simulation. The website link for the python code is provided below.

[Secondary structure of protein during an MD simulation using Biotite] ([Secondary structure during an MD simulation — Biotite 0.41.0 documentation ](https://www.biotite-python.org/examples/gallery/structure/trajectory_sse.html))

The error message goes like this,
 sse[idx] = app.get_sse()
ValueError: could not broadcast input array from shape (71,) into shape (70,)

It is working for the lysozyme.xtc (given as example) and other md trajectory files that I have used, but not this one. How do I solve this issue?

**Comment by padix-key** [comment]
Probably this has something to do with residue recognition. Can you provide the template structure file?

**Reply by intomybioverse** [reply]
[template.txt](https://github.com/user-attachments/files/16053325/template.txt)
I tried with the .gro file (produced after equilibriation) and now the error is as follows. 

(biotite_env) (base) sobs@BEE:~/all_my_data/SCR_wtn3m$ python3 trajectory_sse.py
Traceback (most recent call last):
  File "/home/sobs/all_my_data/SCR_wtn3m/trajectory_ss.py", line 49, in <module>
    app.join()
  File "/home/sobs/.local/biotite_env/lib/python3.10/site-packages/biotite/application/application.py", line 63, in wrapper
    return func(*args, **kwargs)
  File "/home/sobs/.local/biotite_env/lib/python3.10/site-packages/biotite/application/localapp.py", line 257, in join
    self.evaluate()
  File "/home/sobs/.local/biotite_env/lib/python3.10/site-packages/biotite/application/dssp/app.py", line 90, in evaluate
    super().evaluate()
  File "/home/sobs/.local/biotite_env/lib/python3.10/site-packages/biotite/application/localapp.py", line 278, in evaluate
    raise SubprocessError(
subprocess.SubprocessError: 'mkdssp' returned with exit code 1: DSSP could not be created due to an error: invalid formatted floating point number '-7.6293945e-06'

Dssp version 2.3.0 is installed but it is not responding. Then I uninstalled and reinstalled it from this site "https://github.com/cmbi/dssp/archive/refs/tags/2.3.0.tar.gz", installed it and still it is showing mkdssp version  3.1.4 . 

.py file: https://www.biotite-python.org/examples/gallery/structure/trajectory_sse.html
I have attached the template file too.

**Comment by padix-key** [comment, marked as answer]
Sorry for the later answer, I was a few days on vacation. The problem is atom naming: The trajectory from the example gallery was created with GROMACS, whose atom naming sometimes deviate from the standard PDB nomenclature. Specifically, the C-terminal `O` and `OXT` are named `O1` and `O2`. However without the `O` DSSP does not recognize the terminal residue as amino acid, and skips it. Hence, the array with the secondary structure elements has one element less in the example:

```python
# DSSP does not assign an SSE to the last residue -> -1
sse = np.empty((traj.shape[0], struc.get_residue_count(traj)-1), dtype='U1')
```

However, in your template the atoms are named correctly. Hence DSSP returns has the *correct* length. Since the `sse` array is still shortened by 1, this gives an exception due to the mismatch.

To fix this you can rename the problematic C-terminal atoms correctly (i.e. `O1 -> O` in the last residue), in case it not already is. Then you can remove the `-1` in the `sse` creation.
